deploy/chart/templates/_helpers.tpl

```diff
@@ -0,0 +1,97 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+
+{{- define "mosquitto.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "mosquitto.name" -}}
+{{- default .Chart.Name .Values.mosquitto.nameOverride | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "mosquitto.fullname" -}}
+{{- $name := default .Chart.Name .Values.mosquitto.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" $name .Release.Name | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "mosquitto.labels.standard" -}}
+app.kubernetes.io/name: {{ include "mosquitto.name" . }}
+helm.sh/chart: {{ include "mosquitto.chart" . }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+app.kubernetes.io/version: {{ required ".Values.mosquitto.imageTag is required!" .Values.mosquitto.imageTag | quote }}
+{{- end -}}
+
+{{- define "mosquitto.labels.matchLabels" -}}
+app.kubernetes.io/name: {{ include "mosquitto.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end -}}
+
+{{- define "mosquitto.capabilities.kubeVersion" -}}
+{{- .Capabilities.KubeVersion.Version -}}
+{{- end -}}
+
+{{- define "mosquitto.capabilities.ingress.apiVersion" -}}
+{{- if semverCompare "<1.14-0" (include "mosquitto.capabilities.kubeVersion" .) -}}

```

Comment:
the last supported version of kube (not EOL) is 1.27, I don't think you need to support 10 year old versions...

Reply:
I am more inclined to keep this, as it is already implemented and maybe there are some unlucky guys that are stuck with some very old k8s versions

Reply:
Up to you! I feel it keeps complexity for something very, very old that is EOL.